src/Microsoft.AspNetCore.Razor.Evolution/BoundAttributeDescriptor.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    /// <summary>
+    /// A metadata class describing a tag helper attribute.
+    /// </summary>
+    public abstract class BoundAttributeDescriptor
+    {
+        protected BoundAttributeDescriptor(string kind)
+        {
+            Kind = kind;
+        }
```

Comment:
Figured that organizing the descriptor API around `Kind` might make sense. If a user wants to create their own type of descriptor then `Kind` is the only way to truly tell a new descriptor apart from a built-in one.